### PR TITLE
Set sysctls from containers.conf

### PR DESCRIPTION
**Pull request by rhatdan**
Fixes: https://github.com/containers/buildah/issues/4291

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

#### What type of PR is this?

> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test 
> /kind feature
> /kind flake
> /kind other

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Comment by rhatdan** [comment]
@containers/podman-maintainers PTAL
@giuseppe @vrothberg @nalind @flouthoc @ashley-cui @umohnani8 PTAL

File: tests/containers_conf.bats [hunk 1]
```diff
@@ -106,3 +106,24 @@ EOF
...
 }
+
+@test "containers.conf network sysctls" {
+    if test "$BUILDAH_ISOLATION" = "chroot" ; then
+        skip "BUILDAH_ISOLATION = $BUILDAH_ISOLATION"
+    fi
+
+    cat >${TEST_SCRATCH_DIR}/containers.conf << EOF
+[containers]
+default_sysctls = [
```
Comment (cdoern) [inline tests/containers_conf.bats:117]: is there a test for ipc sysctls?

**Review by cdoern — APPROVED** [review]
LGTM, one question

File: tests/containers_conf.bats (hunk 1 above)
Comment (rhatdan) [reply tests/containers_conf.bats:117]: No.

File: run_linux.go [hunk 2]
```diff
@@ -605,11 +605,28 @@ func runMakeStdioPipe(uid, gid int) ([][]int, error) {
...
+
 	// Set namespace options in the container configuration.
 	configureUserns := false
 	specifiedNetwork := false
 	for _, namespaceOption := range namespaceOptions {
 		switch namespaceOption.Name {
+		case string(specs.IPCNamespace):
+			if !namespaceOption.Host {
+				for _, sysctl := range defaultContainerConfig.Sysctls() {
+					splitn := strings.SplitN(sysctl, "=", 2)
```
Comment (flouthoc) [inline run_linux.go:621]: Not a blocker but: both processing sysctls for IPC and Network are using common parsing code there is a room to create a function in `internal/parse` called `parseSysctl(token)` and remove the redundant parsing code.

**Review by flouthoc — COMMENTED** [review]
One nit above otherwise LGTM

File: run_linux.go [hunk 3]
```diff
@@ -684,8 +701,18 @@ func setupNamespaces(logger *logrus.Logger, g *generate.Generator, namespaceOpti
...
 		}
 	}
-	if configureNetwork && !unshare.IsRootless() {
-		for name, val := range define.DefaultNetworkSysctl {
+	if configureNetwork {
+		netSysctls := define.DefaultNetworkSysctl
+		for _, sysctl := range defaultContainerConfig.Sysctls() {
+			splitn := strings.SplitN(sysctl, "=", 2)
+			if len(splitn) > 2 {
+				return false, nil, false, fmt.Errorf("sysctl %q defined in containers.conf must be formated name=value", sysctl)
```
Comment (nalind) [inline run_linux.go:709]:
```suggestion
				return false, nil, false, fmt.Errorf("sysctl %q defined in containers.conf must be formatted as name=value", sysctl)
```

File: run_linux.go [hunk 4]
```diff
@@ -605,11 +605,28 @@ func runMakeStdioPipe(uid, gid int) ([][]int, error) {
...
 	configureUserns := false
 	specifiedNetwork := false
 	for _, namespaceOption := range namespaceOptions {
 		switch namespaceOption.Name {
+		case string(specs.IPCNamespace):
+			if !namespaceOption.Host {
+				for _, sysctl := range defaultContainerConfig.Sysctls() {
+					splitn := strings.SplitN(sysctl, "=", 2)
+					if len(splitn) > 2 {
+						return false, nil, false, fmt.Errorf("sysctl %q defined in containers.conf must be formated name=value", sysctl)
```
Comment (nalind) [inline run_linux.go:623]:
```suggestion
						return false, nil, false, fmt.Errorf("sysctl %q defined in containers.conf must be formatted as name=value", sysctl)
```

File: run_linux.go [hunk 5]
```diff
@@ -605,11 +605,28 @@ func runMakeStdioPipe(uid, gid int) ([][]int, error) {
...
 	for _, namespaceOption := range namespaceOptions {
 		switch namespaceOption.Name {
+		case string(specs.IPCNamespace):
+			if !namespaceOption.Host {
+				for _, sysctl := range defaultContainerConfig.Sysctls() {
+					splitn := strings.SplitN(sysctl, "=", 2)
+					if len(splitn) > 2 {
+						return false, nil, false, fmt.Errorf("sysctl %q defined in containers.conf must be formatted name=value", sysctl)
+					}
+					if strings.HasPrefix(splitn[0], "ipc") {
```
Comment (nalind) [inline run_linux.go:625]: Would this ever happen?

Comment (rhatdan) [reply run_linux.go]: Fixed

**Comment by nalind** [comment]
This looks more or less like what podman's doing in the same area, so LGTM, though testing IPC and UTS sysctls wouldn't hurt.

**Pull request merged**